File: uealib/src/taboosearch/Context.java
package taboosearch;

public class Context extends core.Context {
	
	private static Context instance;

	/**
	 * Возвращает экземпляр контекста.
	 */
	public static Context getInstance() {
		if (instance == null) {
			instance = new Context();
		}
		return instance;
	}
	
	private int ticks = 0;
	
	public void tick() {
		this.ticks++;
		tr.tick();
	}
	
	public int getTicks() {
		return ticks;
	}
	
	Evaluator e;
	Taboolator tr;
	
}

File: taboosearch/src/core/Context.scala
package core
import scala.annotation.unchecked.uncheckedVariance

class Context[+S <: Solution](val evaluator : Evaluator[S],
							  var bestSolutionEver : S @uncheckedVariance,
							  var bestSolutionEverFitness : Double) {

	var ticks : Int = 0;
	def tick() = {
		ticks += 1;
	}
	
}